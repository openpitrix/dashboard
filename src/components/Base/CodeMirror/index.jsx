import React, { lazy, Suspense } from 'react';
import PropTypes from 'prop-types';

const CodeMirror = lazy(() => import('react-codemirror'));

// fixme
if (process.browser) {
  require('codemirror/mode/yaml/yaml');
  require('codemirror/mode/javascript/javascript');
  require('codemirror/lib/codemirror.css');
}

export default class CodeMirrorX extends React.Component {
  static propTypes = {
    code: PropTypes.string,
    mode: PropTypes.string,
    onChange: PropTypes.func
  };

  static defaultProps = {
    code: '',
    onChange: () => {},
    mode: 'yaml'
  };

  constructor(props) {
    super(props);
    this.state = { module: null };
  }

  render() {
    const {
      onChange, code, mode, ...rest
    } = this.props;

    return (
      <Suspense fallback={<div>Loading...</div>}>
        <CodeMirror
          value={code}
          onChange={onChange}
          options={{ mode, lineNumbers: true }}
          {...rest}
        />
      </Suspense>
    );
  }
}
