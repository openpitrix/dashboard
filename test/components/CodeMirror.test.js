import CodeMirror from 'components/CodeMirror';

describe('Component CodeMirror', () => {
  it('basic render', () => {
    const wrapper = <CodeMirror />;

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
