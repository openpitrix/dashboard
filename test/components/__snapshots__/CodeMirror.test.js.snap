// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CodeMirror basic render 1`] = `undefined`;
